sdk/src/main/java/com/google/cloud/dataflow/sdk/io/BoundedSource.java

```diff
@@ -133,6 +133,95 @@ public Double getFractionConsumed() {
     }
 
     /**
+     * A constant to use as the return value for {@link #getParallelismConsumed()} or
+     * {@link #getParallelismRemaining()} when the exact value is unknown.
+     */
+    public static final long PARALLELISM_UNKNOWN = -1;
+
+    /**
+     * Returns the total amount of parallelism in the consumed (and processed) range of this
+     * reader's current {@link BoundedSource} (as would be returned by {@link #getCurrentSource}).
```

Comment:
I'd suggest to explain the difference between consumed and processed here, or refer only to processed. An unprepared reader might assume that these are interchangeable, but they aren't.

I suggest the following definitions:
- A record is "returned" (consumed?) via a call to start/advance that returns true.
- A record is "processed" when the next call to start/advance is made, because we assume that processing happens via a read/process/read/process/.../EOF loop.